### DISCUSSION TITLE
how to use API to add an attachment to field in a form

### DISCUSSION THREAD

**Discussion by mathob** [CSCfi/rems#3252] [General]
I have a form which has a field of type attachment. I have been unable to figure out how to use the API to upload a file to this field. I've tried post_api_applications_save_draft and post_api_applications_add_attachment without success and would appreciate any guidance with which API service to use, and how to specify a call with e.g. curl or Python

Thanks!

**Comment by Macroz** [comment]
Hey,

I assume you are talking about the regular attachment field and its answer. Many commands have also the possibility of a comment or attachment. That is similar but slightly different.

To attach, one must do the following:
-  First the attachment  is uploaded using `POST https://.../api/applications/add-attachment?application-id=123` with `multipart/form-data`  content type, and the file in the `file` parameter. This is a pretty standard file upload. The response contains the `id` of the stored attachment.
- Then the attachment is set as an answer to a field in the application with `POST https://.../api/applications/save-draft` with the field values so that the attachment `id` is set as the value of the correct form/field.
- Now opening the application, it should show the new attachment as the answer to the chosen attachment field.

You can use a test instance of REMS and see what the web application does. If you open the dev console from the browser, and the network tab, you can have a look at what it sends. These can be saved as a curl command etc. too.

Let me know if you can't get it to work.

**Reply by Macroz** [reply]
We have a related issue about improving this so that only one API call is required.

https://github.com/CSCfi/rems/issues/1581

**Comment by mathob** [comment]
YESSSS!!

Thanks again.
